gensim/test/test_corpora.py

```diff
@@ -229,9 +229,107 @@ def test_load(self):
+        it = iter(self.corpus)
+        self.assertEqual(next(it), [(0, 1.0), (1, 1.0), (2, 1.0)])
+        self.assertEqual(next(it), [])
+        self.assertEqual(next(it), [(2, 0.42371910849), (5, 0.6625174), (7, 1.0), (8, 1.0)])
+
+        # confirm that accessing document by index fails
+        self.assertRaises(RuntimeError, lambda: self.corpus[3])
+
+
+class TestMmCorpusNoIndexGzip(CorpusTestCase):
+    def setUp(self):
+        self.corpus_class = mmcorpus.MmCorpus
+        self.corpus = self.corpus_class(datapath('test_mmcorpus_no_index.mm.gz'))
+        self.file_extension = '.mm'
+
+    def test_serialize_compressed(self):
+        # MmCorpus needs file write with seek => doesn't support compressed output (only input)
+        pass
+
+    def test_load(self):
+        self.assertEqual(self.corpus.num_docs, 9)
+        self.assertEqual(self.corpus.num_terms, 12)
+        self.assertEqual(self.corpus.num_nnz, 28)
+
+        # confirm we can iterate and that document values match expected for first three docs
+        it = iter(self.corpus)
+        self.assertEqual(next(it), [(0, 1.0), (1, 1.0), (2, 1.0)])
+        self.assertEqual(next(it), [])
+        self.assertEqual(next(it), [(2, 0.42371910849), (5, 0.6625174), (7, 1.0), (8, 1.0)])
+
+        # confirm that accessing document by index fails
+        self.assertRaises(RuntimeError, lambda: self.corpus[3])
+
+
+class TestMmCorpusNoIndexBzip(CorpusTestCase):
+    def setUp(self):
+        self.corpus_class = mmcorpus.MmCorpus
+        self.corpus = self.corpus_class(datapath('test_mmcorpus_no_index.mm.bz2'))
+        self.file_extension = '.mm'
+
+    def test_serialize_compressed(self):
+        # MmCorpus needs file write with seek => doesn't support compressed output (only input)
+        pass
+
+    def test_load(self):
+        self.assertEqual(self.corpus.num_docs, 9)
+        self.assertEqual(self.corpus.num_terms, 12)
+        self.assertEqual(self.corpus.num_nnz, 28)
+
+        # confirm we can iterate and that document values match expected for first three docs
+        it = iter(self.corpus)
+        self.assertEqual(next(it), [(0, 1.0), (1, 1.0), (2, 1.0)])
+        self.assertEqual(next(it), [])
+        self.assertEqual(next(it), [(2, 0.42371910849), (5, 0.6625174), (7, 1.0), (8, 1.0)])
+
+        # confirm that accessing document by index fails
+        self.assertRaises(RuntimeError, lambda: self.corpus[3])
+
+
+class TestMmCorpusCorrupt(CorpusTestCase):
+    def setUp(self):
+        self.corpus_class = mmcorpus.MmCorpus
+        self.corpus = self.corpus_class(datapath('test_mmcorpus_corrupt.mm'))
+        self.file_extension = '.mm'
+
+    def test_serialize_compressed(self):
+        # MmCorpus needs file write with seek => doesn't support compressed output (only input)
+        pass
+
+    def test_load(self):
+        self.assertRaises(ValueError, lambda: [doc for doc in self.corpus])
```

Comment:
I don't catch, what is it?

Reply:
Are you asking specifically about TestMmCorpusCorrupt?  That's a test to confirm the code will raise an exception if the input file is malformed.  The new code checks that sscanf reads in 3 values, so I just test that and raise an exception if fewer values are read.

As for the other tests, I added tests for:
- the two compressed formats (gzip and bz2)
- to make sure you can access docs by index for a corpus with an index
- to check the values read back match what we expect (mostly, I was worried about floats getting truncated to ints)